core/commands/storage.go

```diff
@@ -807,20 +808,60 @@ func SignContractAndCheckPayment(shardInfo *storage.Shards, shardHash string, ss
 		log.Error(err)
 		return
 	}
+	// check payment
+	cfg, err := cmdenv.GetConfig(env)
+	if err != nil {
+		log.Error(err)
+		return
+	}
+	signedContractID, err := escrow.SignContractID(escrowContract.ContractId, n.PrivateKey)
+	if err != nil {
+		log.Error(err)
+		return
+	}
+	paidIn := make(chan bool)
+	checkPaymentFromClient(req.Context, paidIn, signedContractID, cfg)
+
 	// TODO: download file from client
-	go downloadChunkFromClient(shardInfo, shardHash, ssID, n, pid, req, env)
+	go downloadChunkFromClient(paidIn, shardInfo, shardHash, ssID, n, pid, req, env)
 }
 
 // call escrow service to check if payment is received or not
-func periodicallyCheckPaymentFromClient() {
-	var isReceivedWithTimeOut bool
-	// TODO: isReceivedWithTimeOut := escrow.pay()
-	if !isReceivedWithTimeOut {
-		//TODO: delete file
-	}
+func checkPaymentFromClient(ctx context.Context, paidIn chan bool, contractID *escrowPb.SignedContractID, configuration *config.Config) {
+	timeout := 3 * time.Second
+	newCtx, _ := context.WithTimeout(ctx, timeout)
+	ticker := time.NewTicker(300 * time.Millisecond)
+	paid := false
+	var err error
+
+	go func() {
+		for {
+			select {
+			case t := <-ticker.C:
+				log.Debug("Tick at", zap.Any("time", t.UTC()))
+				paid, err = escrow.IsPaidin(configuration, contractID)
+				if err != nil {
+					log.Error("call IsPaid failed", zap.Error(err))
+				}
+				if paid {
+					paidIn <- true
```

Comment:
need to return `paid` even if false.

Reply:
I mean, once paid, this routine should end.

Reply:
fixed